### PR TITLE
Complete initial port to N-API

### PR DESCRIPTION
**Pull request by NickNaso**
Initial work to support the new N-API as reported here:
[https://github.com/wadey/node-microtime/issues/56](https://github.com/wadey/node-microtime/issues/56)

### CODE REVIEW

File: src/microtime.cc [hunk 1]
```diff
@@ -39,57 +39,57 @@ int gettimeofday(struct timeval *tv, struct timezone *tz) {
...
 #endif
 
-void Now(const Nan::FunctionCallbackInfo<v8::Value> &info) {
+Napi::Value Now(const Napi::CallbackInfo &info) {
   timeval t;
   int r = gettimeofday(&t, NULL);
 
   if (r < 0) {
-    return Nan::ThrowError(Nan::ErrnoException(errno, "gettimeofday"));
+    throw Napi::Error::New(info.Env(), "gettimeofday");
```
Comment (wadey) [inline src/microtime.cc]: The old code here retained the `errno` error code from C land in the error message, does the new Napi::Error does this automatically or is there a way we can preserve that behavior from nan?

Comment (NickNaso) [reply src/microtime.cc]: Hi @wadey,
you are right so I just updated the interested code. Now I create the error and than I set the code field like reported here:

```cpp
Napi::Error e = Napi::Error::New(info.Env(), "gettimeofday");
e.Set("code", Napi::Number::New(info.Env(), errno));
throw e;
```

**Comment by wadey** [comment]
I've fixed up the errno exception a bit more, can I push the improved version directly to your branch so it's part of this PR?

**Comment by NickNaso** [comment]
Hi @wadey,
for me it's ok.

**Comment by wadey** [comment]
Ok, I have pushed my patch for a better ErrnoException. Let me read up on the deployment process for napi and I will merge this soon. Thanks!

**Comment by NickNaso** [comment]
Thanks to you!

**Comment by NickNaso** [comment]
Hi @wadey,
when you publish the tagged version of the module please send me a notice or comment on the related issue.
Thanks!

**Comment by gabrielschulhof** [comment]
@wadey have you had a chance to take a look at this N-API port?

**Comment by wadey** [comment]
This was merged into my https://github.com/wadey/node-microtime/tree/napi branch, which will soon be merged into master. Thank you!

**Pull request closed without merging**